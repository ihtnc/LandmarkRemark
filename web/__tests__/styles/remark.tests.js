import React from 'react';
import { shallow } from 'enzyme';
import {
  RemarkWrapper,
  RemarkHeader,
  RemarkStatus,
  RemarkLabel,
  RemarkInput,
  RemarkButtonWrapper,
  RemarkButton
} from "@styles/remark";

describe("Remark styled components", () => {
  test("RemarkWrapper should match latest snapshot", () => {
    const wrapper = shallow(<RemarkWrapper />)
    expect(wrapper).toMatchSnapshot();
  });

  test("RemarkHeader should match latest snapshot", () => {
    const wrapper = shallow(<RemarkHeader />)
    expect(wrapper).toMatchSnapshot();
  });

  describe("RemarkStatus", () => {
    test("should render correctly for error", () => {
      const wrapper = shallow(<RemarkStatus error={true} />);
      expect(wrapper).toMatchSnapshot();
    });

    test("should render correctly for non error", () => {
      const wrapper = shallow(<RemarkStatus />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  test("RemarkLabel should match latest snapshot", () => {
    const wrapper = shallow(<RemarkLabel />)
    expect(wrapper).toMatchSnapshot();
  });

  test("RemarkInput should match latest snapshot", () => {
    const wrapper = shallow(<RemarkInput />)
    expect(wrapper).toMatchSnapshot();
  });

  test("RemarkButtonWrapper should match latest snapshot", () => {
    const wrapper = shallow(<RemarkButtonWrapper />)
    expect(wrapper).toMatchSnapshot();
  });

  test("RemarkButton should match latest snapshot", () => {
    const wrapper = shallow(<RemarkButton />)
    expect(wrapper).toMatchSnapshot();
  });
});