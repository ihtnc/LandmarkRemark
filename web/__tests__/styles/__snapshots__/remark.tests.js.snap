// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remark styled components RemarkButton should match latest snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bZQynM",
        "isStatic": true,
        "rules": Array [
          "
  min-width: 80px;
  padding: 1vh 1vw;
  margin: 1vh 1vw 1vh 0;
  font-size: 1.5vmax;
  align-self: center;
",
        ],
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bZQynM",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Remark styled components RemarkButtonWrapper should match latest snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-EHOje",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  justify-content: flex-end;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-EHOje",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Remark styled components RemarkHeader should match latest snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": true,
        "rules": Array [
          "
  font-weight: bold;
  padding: 1vh 1vw;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Remark styled components RemarkInput should match latest snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-ifAKCX",
        "isStatic": true,
        "rules": Array [
          "
  font-size: larger;
  padding: 1vh 1vw;
  margin: 1vh 1vw;
  border: 1px solid;
",
        ],
      },
      "displayName": "styled.input",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-ifAKCX",
      "target": "input",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Remark styled components RemarkLabel should match latest snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": true,
        "rules": Array [
          "
  font-size: larger;
  padding: 1vh 1vw;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Remark styled components RemarkStatus should render correctly for error 1`] = `
<StyledComponent
  error={true}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": false,
        "rules": Array [
          "
  padding: 1vh 1vw;
  margin: 1vh 1vw;
  font-size: smaller;
  color: ",
          [Function],
          ";
  align-self: flex-start;
  display: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.span",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Remark styled components RemarkStatus should render correctly for non error 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": false,
        "rules": Array [
          "
  padding: 1vh 1vw;
  margin: 1vh 1vw;
  font-size: smaller;
  color: ",
          [Function],
          ";
  align-self: flex-start;
  display: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.span",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Remark styled components RemarkWrapper should match latest snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  font-size: 1.75vmax;
  display: flex;
  flex-direction: column;
  align-items: stretch;
  min-width: 200px;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
